internal/imageprovider/wikipedia.go

```diff
@@ -163,47 +167,40 @@ func (l *wikiMediaProvider) queryAuthorInfo(thumbnailURL string) (*wikiMediaAuth
 	}, nil
 }
 
-// extractArtistInfo tries to extract the author information as best as possible
-// from the given input which may consist of nested html tags
+// extractArtistInfo tries to extract the author information from the given HTML string.
+// It parses the HTML and attempts to find the most relevant link and text.
 func extractArtistInfo(htmlStr string) (href, text string, err error) {
-	// Parse the HTML string into an HTML document
 	doc, err := html.Parse(strings.NewReader(htmlStr))
 	if err != nil {
 		return "", "", err
 	}
 
-	// Find all the links in the document
 	links := findLinks(doc)
 
-	// If no links are found, extract the inner text and return it
 	if len(links) == 0 {
 		return "", html2text.HTML2Text(htmlStr), nil
 	}
 
-	// If there is only one link, extract the href and inner text and return them
 	if len(links) == 1 {
 		link := links[0]
 		href = extractHref(link)
 		text = extractText(link)
 		return href, text, nil
 	}
 
-	// Look for a Wikipedia user link and extract the href and inner text
 	wikipediaUserLinks := findWikipediaUserLinks(links)
 
 	if len(wikipediaUserLinks) == 0 {
 		return "", "", fmt.Errorf("failed to extract link from HTML: %s", htmlStr)
 	}
 
 	if len(wikipediaUserLinks) == 1 {
-		// Return the href and inner text of the Wikipedia user link
 		wikipediaLink := wikipediaUserLinks[0]
 		href = extractHref(wikipediaLink)
 		text = extractText(wikipediaLink)
 		return href, text, nil
 	}
 
-	// Check if all the links have the same href value
```

Comment:
All of these comments might also be useful to detangle the reason for the differerent if cases